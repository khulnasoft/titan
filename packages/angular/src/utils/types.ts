import type { ProjectConfiguration } from '@titan/devkit';

export type AngularProjectConfiguration = ProjectConfiguration & {
  prefix?: string;
};
