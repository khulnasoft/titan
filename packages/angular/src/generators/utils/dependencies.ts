import { addDependenciesToPackageJson, Tree } from '@titan/devkit';
import { versions } from './version-utils';

export function addBuildableLibrariesPostCssDependencies(tree: Tree): void {
  const pkgVersions = versions(tree);
  addDependenciesToPackageJson(
    tree,
    {},
    {
      postcss: pkgVersions.postcssVersion,
      autoprefixer: pkgVersions.autoprefixerVersion,
      'postcss-url': pkgVersions.postcssUrlVersion,
    }
  );
}
