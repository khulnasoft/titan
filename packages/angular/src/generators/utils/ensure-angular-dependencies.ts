import type { GeneratorCallback, Tree } from '@nx/devkit';
import {
  addDependenciesToPackageJsonIfDontExist,
  getInstalledPackageVersion,
  versions,
} from './version-utils';

export function ensureAngularDependencies(tree: Tree): GeneratorCallback {
  const pkgVersions = versions(tree);

  const angularVersion =
    getInstalledPackageVersion(tree, '@angular/core') ??
    pkgVersions.angularVersion;
  const angularDevkitVersion =
    getInstalledPackageVersion(tree, '@angular-devkit/build-angular') ??
    pkgVersions.angularDevkitVersion;
  const rxjsVersion =
    getInstalledPackageVersion(tree, 'rxjs') ?? pkgVersions.rxjsVersion;
  const tsLibVersion =
    getInstalledPackageVersion(tree, 'tslib') ?? pkgVersions.tsLibVersion;
  const zoneJsVersion =
    getInstalledPackageVersion(tree, 'zone.js') ?? pkgVersions.zoneJsVersion;

  return addDependenciesToPackageJsonIfDontExist(
    tree,
    {
      '@angular/animations': angularVersion,
      '@angular/common': angularVersion,
      '@angular/compiler': angularVersion,
      '@angular/core': angularVersion,
      '@angular/forms': angularVersion,
      '@angular/platform-browser': angularVersion,
      '@angular/platform-browser-dynamic': angularVersion,
      '@angular/router': angularVersion,
      rxjs: rxjsVersion,
      tslib: tsLibVersion,
      'zone.js': zoneJsVersion,
    },
    {
      '@angular/cli': angularDevkitVersion,
      '@angular/compiler-cli': angularVersion,
      '@angular/language-service': angularVersion,
      '@angular-devkit/build-angular': angularDevkitVersion,
      '@angular-devkit/schematics': angularDevkitVersion,
      '@schematics/angular': angularDevkitVersion,
    }
  );
}
