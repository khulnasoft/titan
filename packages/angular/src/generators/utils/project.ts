import type { Tree } from '@titan/devkit';
import { readProjectConfiguration } from '@titan/devkit';
import type { AngularProjectConfiguration } from '../../utils/types';

export function getProjectPrefix(
  tree: Tree,
  project: string
): string | undefined {
  return (
    readProjectConfiguration(tree, project) as AngularProjectConfiguration
  ).prefix;
}
