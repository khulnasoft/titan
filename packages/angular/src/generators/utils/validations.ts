import type { Tree } from '@titan/devkit';
import { getProjects } from '@titan/devkit';

export function validateProject(tree: Tree, projectName: string): void {
  const projects = getProjects(tree);

  if (!projects.has(projectName)) {
    throw new Error(
      `Project "${projectName}" does not exist! Please provide an existing project name.`
    );
  }
}
