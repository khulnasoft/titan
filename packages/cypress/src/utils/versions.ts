export const nxVersion = require('../../package.json').version;
export const eslintPluginCypressVersion = '^2.13.4';
export const typesNodeVersion = '18.16.9';
export const cypressViteDevServerVersion = '^2.2.1';
export const cypressVersion = '^13.6.6';
export const cypressWebpackVersion = '^2.0.0';
export const webpackHttpPluginVersion = '^5.5.0';
export const viteVersion = '~5.0.0';
export const htmlWebpackPluginVersion = '^5.5.0';
