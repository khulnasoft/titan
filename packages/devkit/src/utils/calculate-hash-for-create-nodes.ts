import type { CreateNodesContext } from 'nx/src/devkit-exports';
import { requireNx } from '../../nx';
import { join } from 'path';
const { hashWithWorkspaceContext, hashArray, hashObject } = requireNx();

export function calculateHashForCreateNodes(
  projectRoot: string,
  options: object,
  context: CreateNodesContext,
  additionalGlobs: string[] = []
): string {
  return hashArray([
    hashWithWorkspaceContext(context.workspaceRoot, [
      join(projectRoot, '**/*'),
      ...additionalGlobs,
    ]),
    hashObject(options),
  ]);
}
