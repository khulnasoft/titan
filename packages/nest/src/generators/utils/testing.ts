import { addProjectConfiguration, Tree } from '@titan/devkit';
import { createTreeWithEmptyWorkspace } from '@titan/devkit/testing';

export function createTreeWithNestApplication(appName: string): Tree {
  const tree = createTreeWithEmptyWorkspace();
  addProjectConfiguration(tree, appName, {
    root: `${appName}`,
    sourceRoot: `${appName}/src`,
    projectType: 'application',
    targets: {},
  });
  return tree;
}
