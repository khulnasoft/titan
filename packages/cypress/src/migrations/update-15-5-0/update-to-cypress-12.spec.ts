import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import {
  addProjectConfiguration,
  stripIndents,
  Tree,
  readJson,
} from '@nx/devkit';
import {
  shouldNotOverrideCommands,
  shouldNotUseCyInShouldCB,
  shouldUseCyIntercept,
  shouldUseCySession,
  turnOffTestIsolation,
  updateToCypress12,
} from './update-to-cypress-12';
import { installedCypressVersion } from '../../utils/cypress-version';
jest.mock('../../utils/cypress-version');

describe('Cypress 12 Migration', () => {
  let tree: Tree;
  let mockInstalledCypressVersion: jest.Mock<
    ReturnType<typeof installedCypressVersion>
  > = installedCypressVersion as never;
  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
    jest.resetAllMocks();
  });

  it('should migrate to cy 12', () => {
    mockInstalledCypressVersion.mockReturnValue(11);
    addCypressProject(tree, 'my-app-e2e');
    addCypressProject(tree, 'my-other-app-e2e');
    updateToCypress12(tree);
    assertMigration(tree, 'my-app-e2e');
    assertMigration(tree, 'my-other-app-e2e');
    const pkgJson = readJson(tree, 'package.json');
    expect(pkgJson.devDependencies['cypress']).toEqual('^12.2.0');
  });

  it('should not migrate if cypress version is < 11', () => {
    mockInstalledCypressVersion.mockReturnValue(10);
    addCypressProject(tree, 'my-app-e2e');
    updateToCypress12(tree);
    expect(tree.read('apps/my-app-e2e/cypress.config.ts', 'utf-8'))
      .toMatchInlineSnapshot(`
      "import { defineConfig } from 'cypress';
      import { nxE2EPreset } from '@nrwl/cypress/plugins/cypress-preset';

      export default defineConfig({
        e2e: nxE2EPreset(__filename)
      })"
    `);
  });

  describe('nest cypress commands in should callback', () => {
    beforeEach(() => {
      tree.write(
        'should-callback.ts',
        `describe('something', () => {
  it('should do the thing', () => {
    Cypress.Cookies.defaults()
    cy.server()

    cy.wait('@getApi')
       .its('url').should('include', 'api/v1')
    cy.should((b) => {
      const a = 123;
      // I'm not doing nested cy stuff
    });
    cy.should(($s) => {
      cy.task("");
    })
    cy.should(function($el) {
      cy.task("");
    })
  })
})
`
      );
    });
    it('should comment', () => {
      shouldNotUseCyInShouldCB(tree, 'should-callback.ts');
      expect(tree.read('should-callback.ts', 'utf-8')).toMatchInlineSnapshot(`
        "describe('something', () => {
          it('should do the thing', () => {
            Cypress.Cookies.defaults()
            cy.server()

            cy.wait('@getApi')
               .its('url').should('include', 'api/v1')
            cy.should((b) => {
              const a = 123;
              // I'm not doing nested cy stuff
            });
            /**
        * TODO(@nrwl/cypress): Nesting Cypress commands in a should assertion now throws.
        * You should use .then() to chain commands instead.
        * More Info: https://docs.cypress.io/guides/references/migration-guide#-should
        **/
        cy.should(($s) => {
              cy.task("");
            })
            /**
        * TODO(@nrwl/cypress): Nesting Cypress commands in a should assertion now throws.
        * You should use .then() to chain commands instead.
        * More Info: https://docs.cypress.io/guides/references/migration-guide#-should
        **/
        cy.should(function($el) {
              cy.task("");
            })
          })
        })
        "
      `);
    });

    it('should be idempotent', () => {
      const expected = `describe('something', () => {
  it('should do the thing', () => {
    Cypress.Cookies.defaults()
    cy.server()

    cy.wait('@getApi')
       .its('url').should('include', 'api/v1')
    cy.should((b) => {
      const a = 123;
      // I'm not doing nested cy stuff
    });
    /**
* TODO(@nrwl/cypress): Nesting Cypress commands in a should assertion now throws.
* You should use .then() to chain commands instead.
* More Info: https://docs.cypress.io/guides/references/migration-guide#-should
**/
cy.should(($s) => {
      cy.task("");
    })
    /**
* TODO(@nrwl/cypress): Nesting Cypress commands in a should assertion now throws.
* You should use .then() to chain commands instead.
* More Info: https://docs.cypress.io/guides/references/migration-guide#-should
**/
cy.should(function($el) {
      cy.task("");
    })
  })
})
`;
      shouldNotUseCyInShouldCB(tree, 'should-callback.ts');
      expect(tree.read('should-callback.ts', 'utf-8')).toEqual(expected);
      shouldNotUseCyInShouldCB(tree, 'should-callback.ts');
      expect(tree.read('should-callback.ts', 'utf-8')).toEqual(expected);
    });
  });
  describe('banned Cypres.Commands.overwrite', () => {
    beforeEach(() => {
      tree.write(
        'commands.ts',
        `declare namespace Cypress {
  interface Chainable<Subject> {
    login(email: string, password: string): void;
  }
}
//
// -- This is a parent command --
Cypress.Commands.add('login', (email, password) => {
  console.log('Custom command example: Login', email, password);
});
Cypress.Commands.overwrite('find', () => {});
`
      );
    });
    it('should comment', () => {
      shouldNotOverrideCommands(tree, 'commands.ts');
      expect(tree.read('commands.ts', 'utf-8')).toMatchInlineSnapshot(`
        "declare namespace Cypress {
          interface Chainable<Subject> {
            login(email: string, password: string): void;
          }
        }
        //
        // -- This is a parent command --
        Cypress.Commands.add('login', (email, password) => {
          console.log('Custom command example: Login', email, password);
        });
        /**
        * TODO(@nrwl/cypress): This command can no longer be overridden
        * Consider using a different name like 'custom_find'
        * More info: https://docs.cypress.io/guides/references/migration-guide#Cypress-Commands-overwrite
        **/
        Cypress.Commands.overwrite('find', () => {});
        "
      `);
    });

    it('should be idempotent', () => {
      const expected = `declare namespace Cypress {
  interface Chainable<Subject> {
    login(email: string, password: string): void;
  }
}
//
// -- This is a parent command --
Cypress.Commands.add('login', (email, password) => {
  console.log('Custom command example: Login', email, password);
});
/**
* TODO(@nrwl/cypress): This command can no longer be overridden
* Consider using a different name like 'custom_find'
* More info: https://docs.cypress.io/guides/references/migration-guide#Cypress-Commands-overwrite
**/
Cypress.Commands.overwrite('find', () => {});
`;

      shouldNotOverrideCommands(tree, 'commands.ts');
      expect(tree.read('commands.ts', 'utf-8')).toEqual(expected);
      shouldNotOverrideCommands(tree, 'commands.ts');
      expect(tree.read('commands.ts', 'utf-8')).toEqual(expected);
    });
  });
  describe('api removal', () => {
    it('should be idempotent', () => {
      tree.write(
        'my-cool-test.cy.ts',
        `
describe('something', () => {
  it('should do the thing', () => {
    Cypress.Cookies.defaults()
    Cypress.Cookies.preserveOnce('seesion_id', 'remember-token');
    Cypress.blah.abc() 
    Cypress.Server.defaults({
     delay: 500,
     method: 'GET',
   })
   cy.server()
   cy.route(/api/, () => {
      return {
        'test': 'We’ll',
      }
    }).as('getApi')
 
    cy.visit('/index.html')
    cy.window().then((win) => {
      const xhr = new win.XMLHttpRequest
      xhr.open('GET', '/api/v1/foo/bar?a=42')
      xhr.send()
    })

    cy.wait('@getApi')
   .its('url').should('include', 'api/v1')
  })
})
`
      );
      const expected = stripIndents`describe('something', () => {
      it('should do the thing', () => {
        // TODO(@nrwl/cypress): this command has been removed, use cy.session instead. https://docs.cypress.io/guides/references/migration-guide#Command-Cypress-API-Changes
    Cypress.Cookies.defaults()
        // TODO(@nrwl/cypress): this command has been removed, use cy.session instead. https://docs.cypress.io/guides/references/migration-guide#Command-Cypress-API-Changes
    Cypress.Cookies.preserveOnce('seesion_id', 'remember-token');
        Cypress.blah.abc()
        // TODO(@nrwl/cypress): this command has been removed, use cy.intercept instead. https://docs.cypress.io/guides/references/migration-guide#cy-server-cy-route-and-Cypress-Server-defaults
    Cypress.Server.defaults({
         delay: 500,
         method: 'GET',
       })
       // TODO(@nrwl/cypress): this command has been removed, use cy.intercept instead. https://docs.cypress.io/guides/references/migration-guide#cy-server-cy-route-and-Cypress-Server-defaults
    cy.server()
       // TODO(@nrwl/cypress): this command has been removed, use cy.intercept instead. https://docs.cypress.io/guides/references/migration-guide#cy-server-cy-route-and-Cypress-Server-defaults
    cy.route(/api/, () => {
          return {
            'test': 'We’ll',
          }
        }).as('getApi')

        cy.visit('/index.html')
        cy.window().then((win) => {
          const xhr = new win.XMLHttpRequest
          xhr.open('GET', '/api/v1/foo/bar?a=42')
          xhr.send()
        })

        cy.wait('@getApi')
       .its('url').should('include', 'api/v1')
      })
    })`;
      shouldUseCyIntercept(tree, 'my-cool-test.cy.ts');
      shouldUseCySession(tree, 'my-cool-test.cy.ts');
      expect(stripIndents`${tree.read('my-cool-test.cy.ts', 'utf-8')}`).toEqual(
        expected
      );

      shouldUseCyIntercept(tree, 'my-cool-test.cy.ts');
      shouldUseCySession(tree, 'my-cool-test.cy.ts');
      expect(stripIndents`${tree.read('my-cool-test.cy.ts', 'utf-8')}`).toEqual(
        expected
      );
    });
    it('comment on cy.route,cy.server, & Cypress.Server.defaults usage', () => {
      tree.write(
        'my-cool-test.cy.ts',
        `
describe('something', () => {
  it('should do the thing', () => {
    Cypress.Cookies.defaults()
    Cypress.Cookies.preserveOnce('seesion_id', 'remember-token');
    Cypress.blah.abc() 
    Cypress.Server.defaults({
     delay: 500,
     method: 'GET',
   })
   cy.server()
   cy.route(/api/, () => {
      return {
        'test': 'We’ll',
      }
    }).as('getApi')
 
    cy.visit('/index.html')
    cy.window().then((win) => {
      const xhr = new win.XMLHttpRequest
      xhr.open('GET', '/api/v1/foo/bar?a=42')
      xhr.send()
    })

    cy.wait('@getApi')
   .its('url').should('include', 'api/v1')
  })
})
`
      );
      shouldUseCyIntercept(tree, 'my-cool-test.cy.ts');
      expect(tree.read('my-cool-test.cy.ts', 'utf-8')).toMatchInlineSnapshot(`
        "
        describe('something', () => {
          it('should do the thing', () => {
            Cypress.Cookies.defaults()
            Cypress.Cookies.preserveOnce('seesion_id', 'remember-token');
            Cypress.blah.abc() 
            // TODO(@nrwl/cypress): this command has been removed, use cy.intercept instead. https://docs.cypress.io/guides/references/migration-guide#cy-server-cy-route-and-Cypress-Server-defaults
        Cypress.Server.defaults({
             delay: 500,
             method: 'GET',
           })
           // TODO(@nrwl/cypress): this command has been removed, use cy.intercept instead. https://docs.cypress.io/guides/references/migration-guide#cy-server-cy-route-and-Cypress-Server-defaults
        cy.server()
           // TODO(@nrwl/cypress): this command has been removed, use cy.intercept instead. https://docs.cypress.io/guides/references/migration-guide#cy-server-cy-route-and-Cypress-Server-defaults
        cy.route(/api/, () => {
              return {
                'test': 'We’ll',
              }
            }).as('getApi')
         
            cy.visit('/index.html')
            cy.window().then((win) => {
              const xhr = new win.XMLHttpRequest
              xhr.open('GET', '/api/v1/foo/bar?a=42')
              xhr.send()
            })

            cy.wait('@getApi')
           .its('url').should('include', 'api/v1')
          })
        })
        "
      `);
    });

    it('comment on Cypress.Cookies.defaults & Cypress.Cookies.preserveOnce', () => {
      tree.write(
        'my-cool-test.cy.ts',
        `
describe('something', () => {
  it('should do the thing', () => {
    Cypress.Cookies.defaults()
    Cypress.Cookies.preserveOnce('seesion_id', 'remember-token');
    Cypress.blah.abc() 
    Cypress.Server.defaults({
     delay: 500,
     method: 'GET',
   })
   cy.server()

    cy.wait('@getApi')
   .its('url').should('include', 'api/v1')
  })
})
`
      );
      shouldUseCySession(tree, 'my-cool-test.cy.ts');
      expect(tree.read('my-cool-test.cy.ts', 'utf-8')).toMatchInlineSnapshot(`
        "
        describe('something', () => {
          it('should do the thing', () => {
            // TODO(@nrwl/cypress): this command has been removed, use cy.session instead. https://docs.cypress.io/guides/references/migration-guide#Command-Cypress-API-Changes
        Cypress.Cookies.defaults()
            // TODO(@nrwl/cypress): this command has been removed, use cy.session instead. https://docs.cypress.io/guides/references/migration-guide#Command-Cypress-API-Changes
        Cypress.Cookies.preserveOnce('seesion_id', 'remember-token');
            Cypress.blah.abc() 
            Cypress.Server.defaults({
             delay: 500,
             method: 'GET',
           })
           cy.server()

            cy.wait('@getApi')
           .its('url').should('include', 'api/v1')
          })
        })
        "
      `);
    });
  });
  describe('testIsolation', () => {
    it('should be idempotent', () => {
      const content = `
import { defineConfig } from 'cypress';
import { nxE2EPreset } from '@nrwl/cypress/plugins/cypress-preset';

export default defineConfig({
  e2e: {
    ...nxE2EPreset(__filename),
    testIsolation: true,
})
`;
      tree.write('my-cypress.config.ts', content);
      turnOffTestIsolation(tree, 'my-cypress.config.ts');

      expect(tree.read('my-cypress.config.ts', 'utf-8')).toEqual(content);
    });
    it('should add testIsolation: false to the default e2e config', () => {
      tree.write(
        'my-cypress.config.ts',
        `
import { defineConfig } from 'cypress';
import { nxE2EPreset } from '@nrwl/cypress/plugins/cypress-preset';

export default defineConfig({
  e2e: nxE2EPreset(__filename),
})
`
      );
      turnOffTestIsolation(tree, 'my-cypress.config.ts');
      expect(tree.read('my-cypress.config.ts', 'utf-8')).toMatchInlineSnapshot(`
        "
        import { defineConfig } from 'cypress';
        import { nxE2EPreset } from '@nrwl/cypress/plugins/cypress-preset';

        export default defineConfig({
          e2e: {
            ...nxE2EPreset(__filename),
            /**
            * TODO(@nrwl/cypress): In Cypress v12,the testIsolation option is turned on by default. 
            * This can cause tests to start breaking where not indended.
            * You should consider enabling this once you verify tests do not depend on each other
            * More Info: https://docs.cypress.io/guides/references/migration-guide#Test-Isolation
            **/
            testIsolation: false,
          },
        })
        "
      `);
    });

    it('should add testIsolation: false to inline object e2e config', () => {
      tree.write(
        'my-cypress.config.ts',
        `
import { defineConfig } from 'cypress';
import { nxE2EPreset } from '@nrwl/cypress/plugins/cypress-preset';

export default defineConfig({
  e2e: {
    ...nxE2EPreset(__filename),
    video: false
  }
})
`
      );
      turnOffTestIsolation(tree, 'my-cypress.config.ts');
      expect(tree.read('my-cypress.config.ts', 'utf-8')).toMatchInlineSnapshot(`
        "
        import { defineConfig } from 'cypress';
        import { nxE2EPreset } from '@nrwl/cypress/plugins/cypress-preset';

        export default defineConfig({
          e2e: {
            ...nxE2EPreset(__filename),
            video: false,
            /**
            * TODO(@nrwl/cypress): In Cypress v12,the testIsolation option is turned on by default. 
            * This can cause tests to start breaking where not indended.
            * You should consider enabling this once you verify tests do not depend on each other
            * More Info: https://docs.cypress.io/guides/references/migration-guide#Test-Isolation
            **/
            testIsolation: false,
         }
        })
        "
      `);
    });

    it('should add testIsolation: false for a variable e2e config', () => {
      tree.write(
        'my-cypress.config.ts',
        `
import { defineConfig } from 'cypress';
import { nxE2EPreset } from '@nrwl/cypress/plugins/cypress-preset';
const myConfig = {
    ...nxE2EPreset(__filename),
    video: false
  }

export default defineConfig({
  e2e: myConfig,
})
`
      );
      turnOffTestIsolation(tree, 'my-cypress.config.ts');
      expect(tree.read('my-cypress.config.ts', 'utf-8')).toMatchInlineSnapshot(`
        "
        import { defineConfig } from 'cypress';
        import { nxE2EPreset } from '@nrwl/cypress/plugins/cypress-preset';
        const myConfig = {
            ...nxE2EPreset(__filename),
            video: false
          }

        export default defineConfig({
          e2e: {
            ...myConfig,
            /**
            * TODO(@nrwl/cypress): In Cypress v12,the testIsolation option is turned on by default. 
            * This can cause tests to start breaking where not indended.
            * You should consider enabling this once you verify tests do not depend on each other
            * More Info: https://docs.cypress.io/guides/references/migration-guide#Test-Isolation
            **/
            testIsolation: false,
          },
        })
        "
      `);
    });
  });
});

function addCypressProject(tree: Tree, name: string) {
  const targets = {
    e2e: {
      executor: '@nrwl/cypress:cypress',
      options: {
        tsConfig: `apps/${name}/tsconfig.e2e.json`,
        testingType: 'e2e',
        browser: 'chrome',
      },
      configurations: {
        dev: {
          cypressConfig: `apps/${name}/cypress.config.ts`,
          devServerTarget: 'client:serve:dev',
          baseUrl: 'http://localhost:4206',
        },
        watch: {
          cypressConfig: 'apps/client-e2e/cypress-custom.config.ts',
          devServerTarget: 'client:serve:watch',
          baseUrl: 'http://localhost:4204',
        },
      },
      defaultConfiguration: 'dev',
    },
  };
  addProjectConfiguration(tree, name, {
    root: `apps/${name}`,
    sourceRoot: `apps/${name}/src`,
    projectType: 'application',
    targets,
  });
  // testIsolation
  tree.write(
    `apps/${name}/cypress.config.ts`,
    `import { defineConfig } from 'cypress';
import { nxE2EPreset } from '@nrwl/cypress/plugins/cypress-preset';

export default defineConfig({
  e2e: nxE2EPreset(__filename)
})`
  );
  // test Cypress.Commands.Override
  tree.write(
    `apps/${name}/src/support/commands.ts`,
    `declare namespace Cypress {
  interface Chainable<Subject> {
    login(email: string, password: string): void;
  }
}
//
// -- This is a parent command --
Cypress.Commands.add('login', (email, password) => {
  console.log('Custom command example: Login', email, password);
});
Cypress.Commands.overwrite('find', () => {});
`
  );
  // test .should(() => cy.<cmd>)
  tree.write(
    `apps/${name}/src/e2e/callback.spec.ts`,
    `describe('something', () => {
  it('should do the thing', () => {
    Cypress.Cookies.defaults()
    cy.server()

    cy.wait('@getApi')
       .its('url').should('include', 'api/v1')
    cy.should((b) => {
      const a = 123;
      // I'm not doing nested cy stuff
    });
    cy.should(($s) => {
      cy.task("");
    })
    cy.should(function($el) {
      cy.task("");
    })
  })
})`
  );
  tree.write(
    `apps/${name}/src/e2e/intercept-session.spec.ts`,
    `describe('something', () => {
  it('should do the thing', () => {
    Cypress.Cookies.defaults()
    Cypress.Cookies.preserveOnce('seesion_id', 'remember-token');
    Cypress.blah.abc() 
    Cypress.Server.defaults({
     delay: 500,
     method: 'GET',
   })
   cy.server()
   cy.route(/api/, () => {
      return {
        'test': 'We’ll',
      }
    }).as('getApi')
 
    cy.visit('/index.html')
    cy.window().then((win) => {
      const xhr = new win.XMLHttpRequest
      xhr.open('GET', '/api/v1/foo/bar?a=42')
      xhr.send()
    })

    cy.wait('@getApi')
   .its('url').should('include', 'api/v1')
  })
})`
  );
  tree.write(
    `apps/${name}/src/e2e/combo.spec.ts`,
    `describe('something', () => {
  it('should do the thing', () => {
    Cypress.Cookies.defaults()
    Cypress.Cookies.preserveOnce('seesion_id', 'remember-token');
    Cypress.blah.abc() 
    Cypress.Server.defaults({
     delay: 500,
     method: 'GET',
   })
   cy.server()
   cy.route(/api/, () => {
      return {
        'test': 'We’ll',
      }
    }).as('getApi')
 
    cy.visit('/index.html')
    cy.window().then((win) => {
      const xhr = new win.XMLHttpRequest
      xhr.open('GET', '/api/v1/foo/bar?a=42')
      xhr.send()
    })

    cy.wait('@getApi')
   .its('url').should('include', 'api/v1')
    cy.should(($s) => {
      cy.get('@table').find('tr').should('have.length', 3)
})
  })
})`
  );
}

function assertMigration(tree: Tree, name: string) {
  expect(tree.read(`apps/${name}/cypress.config.ts`, 'utf-8')).toContain(
    'testIsolation: false'
  );
  // command overrides
  expect(tree.read(`apps/${name}/src/support/commands.ts`, 'utf-8')).toContain(
    'TODO(@nrwl/cypress): This command can no longer be overridden'
  );
  // test .should(() => cy.<cmd>)
  expect(tree.read(`apps/${name}/src/e2e/callback.spec.ts`, 'utf-8')).toContain(
    'TODO(@nrwl/cypress): Nesting Cypress commands in a should assertion now throws.'
  );
  // use cy.intercept, cy.session
  const interceptSessionSpec = tree.read(
    `apps/${name}/src/e2e/intercept-session.spec.ts`,
    'utf-8'
  );
  expect(interceptSessionSpec).toContain(
    '// TODO(@nrwl/cypress): this command has been removed, use cy.session instead. https://docs.cypress.io/guides/references/migration-guide#Command-Cypress-API-Changes'
  );
  expect(interceptSessionSpec).toContain(
    '// TODO(@nrwl/cypress): this command has been removed, use cy.intercept instead. https://docs.cypress.io/guides/references/migration-guide#cy-server-cy-route-and-Cypress-Server-defaults'
  );
  // intercept,session & callback
  expect(
    tree.read(`apps/${name}/src/e2e/combo.spec.ts`, 'utf-8')
  ).toMatchSnapshot();
}
