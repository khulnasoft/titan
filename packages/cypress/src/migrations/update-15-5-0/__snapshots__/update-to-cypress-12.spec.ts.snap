// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cypress 12 Migration should migrate to cy 12 1`] = `
"describe('something', () => {
  it('should do the thing', () => {
    // TODO(@nrwl/cypress): this command has been removed, use cy.session instead. https://docs.cypress.io/guides/references/migration-guide#Command-Cypress-API-Changes
Cypress.Cookies.defaults()
    // TODO(@nrwl/cypress): this command has been removed, use cy.session instead. https://docs.cypress.io/guides/references/migration-guide#Command-Cypress-API-Changes
Cypress.Cookies.preserveOnce('seesion_id', 'remember-token');
    Cypress.blah.abc() 
    // TODO(@nrwl/cypress): this command has been removed, use cy.intercept instead. https://docs.cypress.io/guides/references/migration-guide#cy-server-cy-route-and-Cypress-Server-defaults
Cypress.Server.defaults({
     delay: 500,
     method: 'GET',
   })
   // TODO(@nrwl/cypress): this command has been removed, use cy.intercept instead. https://docs.cypress.io/guides/references/migration-guide#cy-server-cy-route-and-Cypress-Server-defaults
cy.server()
   // TODO(@nrwl/cypress): this command has been removed, use cy.intercept instead. https://docs.cypress.io/guides/references/migration-guide#cy-server-cy-route-and-Cypress-Server-defaults
cy.route(/api/, () => {
      return {
        'test': 'We’ll',
      }
    }).as('getApi')
 
    cy.visit('/index.html')
    cy.window().then((win) => {
      const xhr = new win.XMLHttpRequest
      xhr.open('GET', '/api/v1/foo/bar?a=42')
      xhr.send()
    })

    cy.wait('@getApi')
   .its('url').should('include', 'api/v1')
    /**
* TODO(@nrwl/cypress): Nesting Cypress commands in a should assertion now throws.
* You should use .then() to chain commands instead.
* More Info: https://docs.cypress.io/guides/references/migration-guide#-should
**/
cy.should(($s) => {
      cy.get('@table').find('tr').should('have.length', 3)
})
  })
})"
`;

exports[`Cypress 12 Migration should migrate to cy 12 2`] = `
"describe('something', () => {
  it('should do the thing', () => {
    // TODO(@nrwl/cypress): this command has been removed, use cy.session instead. https://docs.cypress.io/guides/references/migration-guide#Command-Cypress-API-Changes
Cypress.Cookies.defaults()
    // TODO(@nrwl/cypress): this command has been removed, use cy.session instead. https://docs.cypress.io/guides/references/migration-guide#Command-Cypress-API-Changes
Cypress.Cookies.preserveOnce('seesion_id', 'remember-token');
    Cypress.blah.abc() 
    // TODO(@nrwl/cypress): this command has been removed, use cy.intercept instead. https://docs.cypress.io/guides/references/migration-guide#cy-server-cy-route-and-Cypress-Server-defaults
Cypress.Server.defaults({
     delay: 500,
     method: 'GET',
   })
   // TODO(@nrwl/cypress): this command has been removed, use cy.intercept instead. https://docs.cypress.io/guides/references/migration-guide#cy-server-cy-route-and-Cypress-Server-defaults
cy.server()
   // TODO(@nrwl/cypress): this command has been removed, use cy.intercept instead. https://docs.cypress.io/guides/references/migration-guide#cy-server-cy-route-and-Cypress-Server-defaults
cy.route(/api/, () => {
      return {
        'test': 'We’ll',
      }
    }).as('getApi')
 
    cy.visit('/index.html')
    cy.window().then((win) => {
      const xhr = new win.XMLHttpRequest
      xhr.open('GET', '/api/v1/foo/bar?a=42')
      xhr.send()
    })

    cy.wait('@getApi')
   .its('url').should('include', 'api/v1')
    /**
* TODO(@nrwl/cypress): Nesting Cypress commands in a should assertion now throws.
* You should use .then() to chain commands instead.
* More Info: https://docs.cypress.io/guides/references/migration-guide#-should
**/
cy.should(($s) => {
      cy.get('@table').find('tr').should('have.length', 3)
})
  })
})"
`;
